===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(p11_parser_formats:formats)) at 11 (line 700, column 8),
        Insert: if ({ p11_parser_formats:n$10 == { 0 } }) free(*(p11_parser_formats:formats->elem)) at 10 (line 700, column 8) }
===== Actual patch in diff format =====
@@ -698,6 +698,8 @@ p11_parser_formats (p11_parser *parser,
 		if (func == NULL)
 			break;
 		if (!p11_array_push (formats, func)) {
+			free(formats->elem);
+			free(formats);
 			va_end (va);
 			return_if_reached ();
 		}
